webofneeds/won-owner-webapp/src/main/webapp/app/actions/connections-actions.js

```diff
@@ -301,20 +301,49 @@ export function connectionsOpen(connectionUri, textMessage) {
   };
 }
 
-export function connectionsConnectAdHoc(theirNeedUri, textMessage) {
+export function connectionsConnectAdHoc(theirNeedUri, textMessage, persona) {
   return (dispatch, getState) =>
-    connectAdHoc(theirNeedUri, textMessage, dispatch, getState); // moved to separate function to make transpilation work properly
+    connectAdHoc(theirNeedUri, textMessage, persona, dispatch, getState); // moved to separate function to make transpilation work properly
 }
-function connectAdHoc(theirNeedUri, textMessage, dispatch, getState) {
+function connectAdHoc(theirNeedUri, textMessage, persona, dispatch, getState) {
   ensureLoggedIn(dispatch, getState).then(async () => {
     const state = getState();
     const theirNeed = getIn(state, ["needs", theirNeedUri]);
     const adHocDraft = generateResponseNeedTo(theirNeed);
     const nodeUri = getIn(state, ["config", "defaultNodeUri"]);
+
+    // create new need
     const { message, eventUri, needUri } = await buildCreateMessage(
       adHocDraft,
       nodeUri
     );
+
+    // add persona
+    if (persona) {
+      const response = await fetch("rest/action/connect", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify([
+          {
+            pending: false,
+            facet: `${persona}#holderFacet`,
```

Comment:
i think the actual holderFacet or holdableFacet uris are stored within the state... let us not assume from the getgo that the uri will look like this but instead retrieve the uri from the need in the state. hint: need.get("hasFacets") -> retrieves a map with facetUri > facetType so we should  be able to find the correct facetUri if we search for the correct facetType... if this is the case in our other connectAction (the one @peacememories implemented we should change it there aswell)

Reply:
This is only possible for the holderFacet. The holdable Facet of the created need can't be accessed via the state, as it is not in the state yet when the persona connect message is sent.